yarn-project/acir-simulator/src/public/index.test.ts

```diff
@@ -36,7 +37,7 @@ describe('ACIR public execution simulator', () => {
   let circuitsWasm: CircuitsWasm;
   let publicState: MockProxy<PublicStateDB>;
   let publicContracts: MockProxy<PublicContractsDB>;
-  let commitmentsDb: MockProxy<CommitmentsDB>;
+  let commitmentsDb: MockProxy<DBOracle>;
```

Comment:
iirc, DBOracle contains methods related to getting notes from the rpc server that will not be available to the sequencer, hence why the other interface was created. 

I think if we use DBOracle the rpc server and sequencer code paths become a bit more coupled, which may increase the pain at separating them out. Although i might misremember why there are two

Reply:
Think it was from my original exploration for #1200, as it needed the function abis to handle it "proper", but with this being moved to a separate pr, will revert to the `CommitmentsDB` here